idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,3677 @@
+        )
+        self.temperature_ref = Param(
+            initialize=298.15,
+            units=units.K,
+            doc="Reference temperature",
+        )
+        self.qm0 = Param(
+            self.isotherm_components,
+            initialize={"CO2": 1.71, "N2": 0.0},
+            units=units.mol / units.kg,
+            doc="Isotherm parameter",
+        )
+        self.b0 = Param(
+            self.isotherm_components,
+            initialize={"CO2": 1.13e5, "N2": 0.0},
+            units=units.bar**-1,
+            doc="Isotherm parameter",
+        )
+        self.t0 = Param(
+            self.isotherm_components,
+            initialize={"CO2": 0.265, "N2": 0.0},
+            units=units.dimensionless,
+            doc="Isotherm parameter",
+        )
+        self.alpha = Param(
+            self.isotherm_components,
+            initialize={"CO2": 0.601, "N2": 0.0},
+            units=units.dimensionless,
+            doc="Isotherm parameter",
+        )
+        self.X = Param(
+            self.isotherm_components,
+            initialize={"CO2": 4.53, "N2": 0.0},
+            units=units.dimensionless,
+            doc="Isotherm parameter",
+        )
+        self.Qb = Param(
+            self.isotherm_components,
+            initialize={"CO2": 62.2, "N2": 0.0},
+            units=units.kJ / units.mol,
+            doc="Isotherm parameter",
+        )
+
+    def _add_inlet_port(self):
+        """
+        Method to build inlet port object of fixed bed TSA. This method allow
+        the unit model to be connected with flue gas streams (stream composed
+        by N2, CO2, H2O, O2) contained in other IDAES unit models. This method
+        add a constraint to force the fixed bed TSA model to use mole flow
+        rates of N2 and CO2 only:
+        flow_mol_in_total = flow_mol_in["CO2"] + flow_mol_in["N2"]
+
+        """
+        # set up inlet port for fixed bed TSA (exhaust gas stream to TSA inlet)
+        self.flow_mol_in = Var(
+            self.flowsheet().time,
+            self.component_list,
+            initialize=10,
+            # domain=PositiveReals,
+            units=units.mol / units.s,
+            doc="Component mole flow rate at inlet [mol/s]",
+        )
+        self.temperature_in = Var(
+            self.flowsheet().time,
+            initialize=300,
+            domain=PositiveReals,
+            units=units.K,
+            doc="Temperature at inlet [K]",
+        )
+        self.pressure_in = Var(
+            self.flowsheet().time,
+            initialize=101325,
+            domain=PositiveReals,
+            units=units.Pa,
+            doc="Pressure at inlet [Pa]",
+        )
+
+        # create empty port
+        p = Port(noruleinit=True, doc="Inlet port for fixed bed TSA")
+        # add port object as an attribute to model
+        setattr(self, "inlet", p)
+        # dictionary containing state of inlet stream
+        inlet_dict = {
+            "flow_mol_comp": self.flow_mol_in,
```

Comment:
I think you can do this as part of the `Port` declaration as well (probably using the `initialize` argument).

Again, this approach is only needed when you have to procedurally generate the port contents.

Reply:
Now passing `inlet_dict` as `initialize` argument